### PR TITLE
feat(relevantSort): implement canRefine

### PR DESCRIPTION
**Pull request by eunjae-lee**
**Summary**

implements canRefine in connectRelevantSort as true when nbHits > 0 and it's a virtual replica.

**Result**

relevantSort now has a canRefine

### CODE REVIEW

File: src/connectors/relevant-sort/connectRelevantSort.ts [hunk 1]
```diff
@@ -77,11 +93,14 @@ const connectRelevantSort: RelevantSortConnector = function connectRelevantSort(
...
 
+        const isVirtualReplica = appliedRelevancyStrictness !== undefined;
+
         return {
           isRelevantSorted:
             typeof appliedRelevancyStrictness !== 'undefined' &&
             appliedRelevancyStrictness > 0,
-          isVirtualReplica: appliedRelevancyStrictness !== undefined,
+          isVirtualReplica,
+          canRefine: isVirtualReplica && results?.nbHits! > 0,
```
Comment (eunjae-lee) [inline src/connectors/relevant-sort/connectRelevantSort.ts:103]: Even if `nbSortedHits === 0`, there can be a possibility to see different results with different `relevancyStrictness`. So we're checking it with `nbHits`, not with `nbSortedHits`.

Comment (Haroenv) [reply src/connectors/relevant-sort/connectRelevantSort.ts:103]: I wonder if just `isVirtualReplica` isn't enough here, can't you in theory still swap between virtual and regular view of something with 0 hits? On the other hand, having no hits is probably a good condition to hide it

**Review by Haroenv — APPROVED** [review]

File: src/connectors/relevant-sort/connectRelevantSort.ts (hunk 1 above)
Comment (eunjae-lee) [reply src/connectors/relevant-sort/connectRelevantSort.ts:103]: Umm right. We can swap between virtual and regular, so even if it's not virtual, as long as nbHits > 0, it should be able to refine. So I'm thinking of just `results?.nbHits! > 0`. Makes sense to you?

Comment (Haroenv) [reply src/connectors/relevant-sort/connectRelevantSort.ts:103]: actually thought about it again, and I think the current behaviour is correct. Can someone confirm? @francoischalifour @shortcuts

Comment (shortcuts) [reply src/connectors/relevant-sort/connectRelevantSort.ts:103]: > so even if it's not virtual, as long as nbHits > 0, it should be able to refine.

But refine rely on the `relevancyStrictness`, so we should check if it's a virtual replica anyway, no?

I think the current implementation make sense

Comment (eunjae-lee) [reply src/connectors/relevant-sort/connectRelevantSort.ts:103]: Oh right. `isVirtualReplica` is true when it's 0, too. So it seems alright to me.

**Review by shortcuts — APPROVED** [review]

**Review by Haroenv — APPROVED** [review]

**Pull request merged**